### DISCUSSION TITLE
new error: The following families had failure rates

### DISCUSSION THREAD

**Discussion by LiS-88** [hahnlab/CAFE5#96] [Q&A] [answered]
Hello! When I used cafe5, the initial error message was "You may want to try removing the top few families with the largest difference between the max and min counts and then re-run the analysis". Then,I added the -l parameter and set it to l=0.0001.Now,The new error message at the end of the run is  "The following families had failure rates >20% of the time".When I delete that family and run it again, I get a new family of genes that fail.How should this situation be resolved?
My parameters and results are as follows:cafe5 -i cafe.txt -t tree.txt  -l 0.0001 -p -k 3 -o cafe.gamma_results
error:
53 values were attempted (0% rejected)
The following families had failure rates >20% of the time:
OG0000002 had 25 failures

**Comment by benfulton** [comment, marked as answer]
I suppose you're trying to optimize an alpha value? What you are probably seeing is the optimizer attempting various alpha values that are out of range for your data. If a value is calculated at the end there's probably nothing to worry about; otherwise you might try changing the value of k or continuing to exclude the families that CAFE is having difficulty with. (The failures are telling you that the family size may be too big in those clusters, if the variance among families is too high CAFE has a hard time converging on a reasonable value.)

**Reply by Jiahe-Sun** [reply]
Can I understand it as "If CAFE finally successfully outputs a set of parameter values (for example, lambda=0.0012639588911623, alpha=0.79433068553981, -lnL=106905.16948757), Then even if there are non-convergent families, the overall result may still be reliable. That is, although some families do not converge, the optimizer may have found a set of global parameters that reasonably fits most of the data." . What confused me was that even after adjusting different k values, about 60% of gene families did not converge.
